### DISCUSSION TITLE
Graphql Transaction vs. Span

### DISCUSSION THREAD

**Discussion by keithhackbarth** [getsentry/sentry-python#3196] [Q&A] [answered]
What is the logic for span vs. transactions for the graqhql integrations like Strawberry and Graphene? For example, I would think a single graphql operation (eg `GetUsers`) would be represented as a transaction so we could see which operations are trending on the Performance dashboard. Instead, we treat it as a view and then the result is that all the dashboard shows is the root view `/graphql` which does not yield much value.

Happy to submit a PR but before I do - wanted to understand the use case and logic behind this decision?

**Comment by czyber** [comment, marked as answer]
Hey @keithhackbarth 

We got the same problem before.

So it seems the issue you are having (with Strawberry) should be fixed by #3294 .
Graphene on the other hand still has the behaviour you are describing.

As the changes are already made for Strawberry I dont think anything would speak against submitting a PR for graphene as well.
If for some reasons that would not be accepted, you could get the desired functionality by implementing it as a [Middleware](https://docs.graphene-python.org/en/latest/execution/middleware/) for Graphene that sets the transaction name.
